test/cljs/status_im/test/utils/contenthash.cljs

```diff
@@ -7,7 +7,7 @@
 
 (deftest contenthash-decode
   (testing "decoding a valid ipfs hash"
-    (is (= (contenthash/decode "0xe301122029f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f")
+    (is (= (contenthash/decode "0xe3010170122029f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f")
```

Comment:
previous test was fine, the extra 01 is a typo in EIP and 70 is certainly not the content type we want for all our decode files

Reply:
Not it's not I double checked with spec writer.

Reply:
did he provide an explanation for the extra 01 then?

Reply:
The second 01 is variable length @yenda

Reply:
@decanus could you point me to the spec that specify that I can't find it. Is it specific to EIP1577? It makes the contenthash unreadable with a tool like ipfs CID inspector https://cid.ipfs.io/#f017010122029f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f

Reply:
providing variable length isn't necessary because multicodecs codes are distributed so that they don't overlap, but introducing a variable length could introduce ambiguity

Reply:
I'm not 100% certain, let me get @arachnid in

Reply:
Per multiformats/multicodec:

> multicodec is a self-describing multiformat, it wraps other formats with a tiny bit of self-description. A multicodec identifier is a varint.

The varint encoding for 0xe3 (IPFS) is 0xe301 - 'e30101' in the EIP1577 example is multicodec value 0xe3, followed by CID version 0x01.

Reply:
@Arachnid why is the the varint encoding done like this? Also it is not consistent between ipfs and swarm examples:

`0x e30101 (ipfs) 70 12 20 (hashing algo and length) 29f2d17be6139079dc48696d1f582a8530eb9805b561eda517e22a892c7e3f1f (hash)`

`0x e40101 (swarm) fa01 (swarm manifest) 1b 20 (hashing algo and length) d1de9994b4d039f6548d191eb26786769f580809256b4685ef316805265ea162 (hash)`

This way of encoding is not compatible with the cid inspector mentionned in the spec https://cid.ipfs.io/#f01701b20d1de9994b4d039f6548d191eb26786769f580809256b4685ef316805265ea162 and does not correspond to the exemples I could find in implementations of multiformat. 

For instance:
- https://github.com/multiformats/js-cid has
`zdj7WkRPAX9o9nb9zPbXzwG7JEs78uyhwbUs8JSUayB98DWWY`
which is
`0x01701220DEBD7ADB3CE56544D22A6F6B93396F6980A8067C2CC134F0F7801B6331092B95`

Reply:
The use of varints is described in the multicodec spec: https://github.com/multiformats/multicodec. Any compliant multicodec encoder/decoder should handle it.

Multicodec values are not CIDs, but these particular ones contain a CID. If you strip off the varint prefix, they should decode as a CID.

The CID you link to appears to be valid. I'm not sure what you're trying to demonstrate with that link?

I'm also not sure what's inconsistent about the IPFS and Swarm examples; can you clarify?

Reply:
@Arachnid Ok so after experimenting more with cid.ipfs.io and the cid-js library I can confirm that I was wrong and the EIP examples are correct and compatible with the different implementations. 

I confused myself because the varint encoding of `70` is `70` and after looking at the protobuf doc https://developers.google.com/protocol-buffers/docs/encoding#varints I understand how it is done and why the encoding of `e3` is `e301`. Thank you for your patience.

Reply:
@Arachnid Thanks for the clarification! Is the intention to allow support for various content-types and hash functions? Or shall we treat the examples as normative?

Reply:
@yenda No worries!

@jeluard Certainly - as many as you care to support. The examples are just the ones we know of so far.

Reply:
@Arachnid in CIDv1 it seems like the content-type isn't optional, however the current codecs for content-type are quite limited, for instance nothings really fits for a dapp in html5. Isn't it going to be a problem for adoption of the EIP?

Reply:
actually it doesn't seem to be a goal of multiformat to support these kind of content-types:

Q. What formats (multicodec codes) does CID support?

`We are figuring this out at this time. It will likely be a table of formats for secure distributed systems. So far, we want to address: IPFS's original protobuf format, the new IPLD CBOR format, git, bitcoin, and ethereum object`

Reply:
To be clear, the content-type field here specifies the type of the hash - in the case of IPFS it's `dag-pb` - not the type of the content it references.

Reply:
ok so if I understand correctly for IPFS contenthashes when we use the infura gateway with the simple `add` method instead of `dag_put` we have `raw 0x00` content-type? And same applies for swarm when uploading a file directly without a manifest?

Reply:
That sounds right - but since IPFS returns CIDs, you should simply be able to use the CID it gives you, and let it worry about the content type.

Reply:
yes that's what I do now, I added the `cid-version` option to our call to `add` so that it returns a b58 encoded CID, then I just have to convert it to hex and prepend "0xe30101" to get the contenthash